app/src/androidTest/java/com/github/h3lp3rs/h3lp/MainPageTestActivity.kt

```diff
@@ -26,26 +34,39 @@ import org.junit.runner.RunWith
 
 @RunWith(AndroidJUnit4::class)
 class MainPageTestActivity {
-
-    @get:Rule
-    val testRule = ActivityScenarioRule(
-        MainPageActivity::class.java
-    )
-
     @get:Rule
     var mRuntimePermissionRule: GrantPermissionRule = GrantPermissionRule.grant(Manifest.permission.ACCESS_FINE_LOCATION)
 
     @Before
     fun setup() {
-        Intents.init()
+        globalContext = getApplicationContext()
+        userUid = USER_TEST_ID
+        PREFERENCES.db = MockDatabase()
+        resetStorage()
+        storageOf(Storages.USER_COOKIE).setBoolean(GUIDE_KEY, true)
+    }
+
+    private fun launch(): ActivityScenario<MainPageActivity> {
+        return launch(Intent(getApplicationContext(), MainPageActivity::class.java))
+    }
+
+    private fun launchAndDo(action: () -> Unit) {
+        launch().use {
+            start()
+            action()
+            end()
+        }
+    }
```

Comment:
This is a really nice refactoring! It could even be (along with the start and end) put in a utils testing file to be reused since it's also used in the medical card activity test, search bar test, and others and could prove useful elsewhere.

Reply:
Definitely agree, I only realized afterwards the number of times I used this. We can name the class TestSetup or something like that. It will contain all required functions to create custom test environments and launch them properly.